fineract-provider/src/main/java/org/apache/fineract/portfolio/client/service/ClientWritePlatformServiceJpaRepositoryImpl.java

```diff
@@ -631,7 +631,7 @@ public CommandProcessingResult closeClient(final Long clientId, final JsonComman
                 throw new InvalidClientStateTransitionException("close", "is.under.transfer", errorMessage);
             }
 
-            if (client.isNotPending() && client.getActivationLocalDate().isAfter(closureDate)) {
+            if (client.isNotPending() && client.getActivationLocalDate() != null && client.getActivationLocalDate().isAfter(closureDate)) {
```

Comment:
@nazeer1100126 
There's no guarantee that the "activationDate" property of all clients will have a value other than NULL or an empty string. During migration of a new tenant, some data might be missed. Moreover, it is good practice to carry out such checks to avoid null pointer exceptions.